### PR TITLE
The final nail in the unfun round removal coffin

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

Round removal is great... if you're not on a server that forbids it most of the time.

Currently, we have to tell players off for using certain methods of round removal(shuttle gibbing), and this is a problem. So why not just remove the method entirely?

This PR does two things:
1. Removes the gibbing aspect of shuttles. Instead, shuttles act much like trams(this is logical) and will yeet said person into oblivion(maybe).
2. Crematoriums have been reworked and no longer instantly qdel the occupant. Instead, they are subject to what a real crematorium would do, an intense inferno, plus some(lots) of pain. It basically slowly burns the thing inside until it's either dead or there are no bodyparts left to burn. Once a bodypart reaches a maximum amount of damage, it disintegrates into ash.
3. Removes shield gen RR by decapping instead of gibbing humans.

This puts these methods of round removal more in line with our server policy.

# But this buffs lings!

## Parasitic extraction

This surgery will enable the RESEARCH DIRECTOR to de-ling a changeling. It takes a bit of time. This should solve this aspect.

Pending head admin input.

## Changelog
:cl:
balance: The crematorium and shuttle have less round removal attached to them.
/:cl:

### CODE REVIEW

**Comment by Gandalf2k15** [comment]
To do: make shield gens cut people in half instead of gibbing. 

Make the gibber not gib humans.

**Comment by Gandalf2k15** [comment]
~To do: replace gib on humans with a fun way to die mechanic that allows for revival still.~ - Not doing this.

**Comment by Gandalf2k15** [comment]
~To do: make holoparas remove the para not the host.~ - Not doing this

**Comment by ForrestWick** [comment]
> To do: make holoparas remove the para not the host.

I feel like holoparas are powerful enough to justify this. Just make sure the description has in all caps that if you die, you dust. Have people understand if they choose it anyway, they're accepting that risk, as it's supposed to be a powerful weapon at the cost of if you mess up and die, it's game over (for that character exclusively that round anyway)

**Comment by FlamingLily** [comment]
> > To do: make holoparas remove the para not the host.
> 
> I feel like holoparas are powerful enough to justify this. Just make sure the description has in all caps that if you die, you dust. Have people understand if they choose it anyway, they're accepting that risk, as it's supposed to be a powerful weapon at the cost of if you mess up and die, it's game over (for that character exclusively that round anyway)

I have to agree with this. As a server, we shouldn't be anti-RR, just anti-forced-RR

Megafauna are fine to RR people, because it's the miner's choice to attack them

Horrorlings are fine to RR the ling on death, because it's the ling's choice to pop it

Holoparas, likewise, are fine to RR the user on death, because it's the tot's choice to buy one.

And at the end of the day, respawns exist.

**Comment by UltimateFluff** [comment]
I just want to voice a bit of a concern that this PR has. While yes, means to remove RR stuff is nice...you're getting on a very slippery slope with this sort of move - you put in the ability to de-ling lings, you're effectively putting in a means to de-antag an antag, and Security has a history of being a bit too...zealous with trying to deal with antags in a "not technically RRing them but they can't do anything short of cryo" fashion. This would effectively give them the means to tackle lings in the same sense as pacifying traitors for existing and so on - the moment you put in a means to de-antag an antag, the slope created will eventually lead to "remove antags period".
The moment we get to that, we might as well rebranch to Polaris and be Not Virgo. So keep this stuff in mind, please, as we still have to deal with getting policies *finished* let alone *hammered out* in a number of places, and if memory serves me correctly, we have plenty of cases where policy isn't exactly followed *due to* not being finished. Just my two cents on the whole matter.

**Comment by FlamingLily** [comment]
As for the actual features of the PR, shuttle-gibbing being removed is fine, in my eyes. Crematorium dusting should probably still exist, however, namely because that's how crematoriums work, and as above, lings shouldn't be able to be de-linged.

**Comment by softcerv** [comment]
> 
> We have a solution to this(WIP), there will be methods of de-linging someone. Instead of RRing them, you will be required to perform surgery to remove the creature from within. It will be moderately hard so that it can replace the whole one button removes ling feature.
> 
> Still debating on how this one should work. Perhaps the above stop revival? Or just delings them.
> 
I know it's a WIP Idea, but please find some way to make this surgery incredibly risky or something so that security doesn't constantly try and have medical preform the de-linging people every time the suspect someone of being a ling. Speaking from my experience in the past with pacification surgery as a sec medic, there will likely be security that will try and abuse this especially if it's low risk to the host.

**Comment by capsaicinz** [comment]
> To do: replace gib on humans with a fun way to die mechanic that allows for revival still.

maybe a similar state to husked, where the entire body is just in a mangled/shredded state, and needs to be repaired a certain way before revival is possible? essentially brute husk

**Comment by Gandalf2k15** [comment]
All valid concerns, I'll address them later

**Comment by thestubborn** [comment]
" you will be required to perform surgery to remove the creature from within"
in the actual film, copper tried to defib norris, who's infected
the paddles touch it's chest and **_the chest_** bites off copper's hands
you cant deling a ling

**Comment by SpookyTheFox** [comment]
One more for the round removal pile, the borg console. It's a crutch meant for low pop servers, and on skyrat only punishes borg players for not immediately rendering the round unplayable. It's also not even your choice to get subverted, you must play it if it happens. So, that one should go too. There's still plenty of un-fun ways to hard stun borgs that all of security and command have access to. I suggest a little bit of UI magic, make it so the buttons on the console don't work.

**Comment by Tbowes-dev** [comment]
To further on the exploding borg console, An emagged or Malf hacked cyborg's explosion is greater but it almost always deletes the Posi or in worse cases the brain of a crew member.
If your aiming to remove round removal, this is just another one to add to your list.

**Comment by thestubborn** [comment]
borg console isnt a crutch
its "we didn't prepare to get locked down"
this is meant to remove RR
not the worry of borg
(and borg is only fun when you get subverted)

**Comment by SpookyTheFox** [comment]
> 
> 
> borg console isnt a crutch
> its "we didn't prepare to get locked down"
> this is meant to remove RR
> not the worry of borg
> (and borg is only fun when you get subverted)

We are discussing round removal, not your opinions.
`/mob/living/silicon/robot/proc/self_destruct()
	if(emagged)
		QDEL_NULL(mmi)`
Try to keep conversation relevant.

**Comment by SpaceVampire** [comment]
> To further on the exploding borg console, An emagged or Malf hacked cyborg's explosion is greater but it almost always deletes the Posi or in worse cases the brain of a crew member.
> If your aiming to remove round removal, this is just another one to add to your list.

It doesn't matter if a cyborg gets deleted because they can just respawn into a new posi-brain.
Which are really pretty free, when it comes to printing.
Impossible to really round remove them, in the same way as actual crew.

**Comment by tf-4** [comment]
-1 code solution to a player issue

**Comment by oilysnake** [comment]
Are you going to remove the gibbing aspect of megafauna, gibtonite or the RR of chasms?

 
> Megafauna are fine to RR people, because it's the miner's choice to attack them
> 

Sometimes Megafauna just aggro from random bullshit like gibtonite exploding 2 miles away or a spare, random-ass plasma bolt hitting a megafauna in a wall

**Comment by BurgerLUA** [comment]
Just do what burgerstation does, partially, and make it so that getting crushed by a shuttle instantly kills you with high brute damage on every limb and makes your mob get "stuck" under the tile until its moved again.

**Comment by hugmeorelse** [comment]
> Just do what burgerstation does, partially, and make it so that getting crushed by a shuttle instantly kills you with high brute damage on every limb and makes your mob get "stuck" under the tile until its moved again.

@Gandalf2k15 I like this idea.

**Comment by YakumoChen** [comment]
Shuttle gib is, per the rules, a 'last resort' to destroy lings. The crematorium can be sabotaged, as can gibbers. Without a way to disable a ling they will continue to revive over and over until they remove all opposition, bar extreme, metagame methods such as BZ anesthetic.

As far as the deling surgery, it appears limited to the RD (or their skillchip, which I don't think is obtainable otherwise). RD often isn't played most rounds, or even assumes they're alive, making it extremely unlikely a deling will ever happen on most ling rounds.

**Comment by Gandalf2k15** [comment]
> > Just do what burgerstation does, partially, and make it so that getting crushed by a shuttle instantly kills you with high brute damage on every limb and makes your mob get "stuck" under the tile until its moved again.
> 
> @Gandalf2k15 I like this idea.

Easy enough to do

**Comment by KathyRyals** [comment]
I don't see the issue with round removal options on a server where you can respawn at any time with any other characters.

**Comment by Seris02** [comment]
I'm starting to agree with this PR, but the surgery definitely shouldn't just be limited to the research director.

**Comment by BurgerLUA** [comment]
Honestly I think personally that gibbing is a blight on the server. I've never been gibbed before so this isn't salt, but I think the concept of unable to being to be revived is bad on a server that had cloning removed. On /tg/ it makes sense because it's a game, but on Skyrat it's more of an ERP or at least a roleplay server. Many people just have one character so they probably aren't too happy about getting round removed because of event RNG or just being in the wrong place at the wrong time.

File: modular_skyrat/master_files/code/game/objects/structures/morgue.dm [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		if(M.stat == DEAD)
+			if(ishuman(M))
+				var/mob/living/carbon/human/departed_soul = M
+				departed_soul.become_husk(src)
+			continue
+		alive_mobs++
+		if(ishuman(M))
+			var/mob/living/carbon/human/burning_soul = M
+			var/list/bodyparts = burning_soul.bodyparts
+			shuffle(bodyparts)
```
Comment (BurgerLUA) [inline modular_skyrat/master_files/code/game/objects/structures/morgue.dm:44]: What's the point of shuffling here?

Comment (Gandalf2k15) [reply modular_skyrat/master_files/code/game/objects/structures/morgue.dm:44]: for random effect

**Comment by hugmeorelse** [comment]
> I'm starting to agree with this PR, but the surgery definitely shouldn't just be limited to the research director.

I asked for it to apply to CMO and MD too.

File: modular_skyrat/master_files/code/game/objects/structures/morgue.dm (hunk 1 above)
Comment (BurgerLUA) [reply modular_skyrat/master_files/code/game/objects/structures/morgue.dm:44]: Yeah but the pick() is already random.

**Comment by ReynTheLord** [comment]
We just had a changeling round. this. bullshit. and TWO HORROR FORMS, in the same round. Not fun.

**Comment by Funce** [comment]
Motion Refused. This will be un-test-merged next round.

**Comment by Lauwma** [comment]
Yes being unable to shuttlegib ling is stupid. If they are gamer lings that don't want to RP I don't see why we shouldn't be allowed to round remove them. Sometimes losing as an antag is fine, losing is part of the game.

**Comment by Echopsychotic** [comment]
Well I was going to ask if this wasnt closed if this solves the miner gibbing problem though oh well.

**Comment by ForrestWick** [comment]
I think the shuttlegib change should still be applied, as its fuckinf stupid, but the stuff pertaining to lings? Ehhhhhhh

**Pull request closed without merging**